pkg/api/folder.go

```diff
@@ -220,11 +220,15 @@ func (hs *HTTPServer) CreateFolder(c *contextmodel.ReqContext) response.Response
 }
 
 func (hs *HTTPServer) setDefaultFolderPermissions(ctx context.Context, orgID int64, user identity.Requester, folder *folder.Folder) error {
-	if !hs.Cfg.RBAC.PermissionsOnCreation("folder") {
+	isNested := folder.ParentUID != ""
+	if (isNested && hs.Features.IsEnabled(ctx, featuremgmt.FlagNestedFolders)) || !hs.Cfg.RBAC.PermissionsOnCreation("folder") {
```

Comment:
Do you even have to check `featuremgmt.FlagNestedFolders` here. I imagine it is not possible to set  `folder.ParentUID`  if this toggle is not enabled. 

```suggestion
	if isNested || !hs.Cfg.RBAC.PermissionsOnCreation("folder") {
```

Reply:
I was wondering about it too, especially as I imagine the FT will be removed soon anyway. But this is the check that we had before, so I decided to just leave it there for now.

Reply:
Ok, fair enough. You're right that we should never be in a situation where parent folder UID is set without the FT enabled.